utils/moduleVisitor.js

```diff
@@ -111,6 +142,17 @@ function makeOptionsSchema(additionalProperties) {
       'commonjs': { 'type': 'boolean' },
       'amd': { 'type': 'boolean' },
       'esmodule': { 'type': 'boolean' },
+      'requireResolve': {
+        'oneOf': [
+          { 'type': 'boolean' },
+          {
+            'type': 'object',
+            'properties': {
+              'commonjs': { 'type': 'boolean' },
```

Comment:
I've made the `requireResolve` option either be a boolean or a shape with a `commonjs` option.

`commonjs: true, requireResolve: true` and `commonjs: false, requireResolve: { commonjs: true }` will both lint `require.resolve`.

As far as I can see `requirejs` doesn't have its own version of `require.resolve`, but perhaps it's better to leave the door open for that?

Reply:
I wouldn't count on AMD getting any updates any time soon; it's effectively dead (even if it's continuing to be maintained).

Reply:
So should I revert this to be just a boolean?